packages/typir/src/kinds/bottom-kind.ts

```diff
@@ -0,0 +1,147 @@
+/******************************************************************************
+ * Copyright 2024 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import { TypeEqualityProblem } from '../features/equality.js';
+import { InferenceRuleNotApplicable } from '../features/inference.js';
+import { SubTypeProblem } from '../features/subtype.js';
+import { isType, Type } from '../graph/type-node.js';
+import { Typir } from '../typir.js';
+import { TypirProblem } from '../utils/utils-definitions.js';
+import { createKindConflict } from '../utils/utils-type-comparison.js';
+import { toArray } from '../utils/utils.js';
+import { isKind, Kind } from './kind.js';
+
+export class BottomType extends Type {
+    override readonly kind: BottomKind;
+
+    constructor(kind: BottomKind, identifier: string) {
+        super(identifier);
+        this.kind = kind;
+    }
+
+    override getUserRepresentation(): string {
+        return this.identifier;
+    }
+
+    override analyzeTypeEqualityProblems(otherType: Type): TypirProblem[] {
+        if (isBottomType(otherType)) {
+            return [];
+        } else {
+            return [<TypeEqualityProblem>{
+                $problem: TypeEqualityProblem,
+                type1: this,
+                type2: otherType,
+                subProblems: [createKindConflict(this, otherType)],
+            }];
+        }
+    }
+
+    override analyzeIsSubTypeOf(_superType: Type): TypirProblem[] {
```

Comment:
The reason is simple and strong in my eyes: The user of the DSL will not be happy about "no" only. The user wants to know, WHY not. From the returned problems, such useful hints can be generated.

Is it a naming issue or do you prefer another API/function in general?

Reply:
I'd say what irritates me most is the name "Problem".

Reply:
Got it now, you are completely right! Since I have no better name, I keep the term "problem" for now, but I added this "problem" issue to our list of open terminology issues.